service/rest_task.go

```diff
@@ -151,13 +151,12 @@ func (restapi restAPI) getTaskInfo(w http.ResponseWriter, r *http.Request) {
 	}
 	destTask.LocalTestResults = make(taskTestResultsByName, len(srcTask.LocalTestResults))
 	for _, tr := range srcTask.LocalTestResults {
-		numSecs := tr.EndTime - tr.StartTime
 		testResult := taskTestResult{
 			Status:    tr.Status,
-			TimeTaken: time.Duration(numSecs * float64(time.Second)),
-			Logs:      taskTestLogURL{tr.URL},
+			TimeTaken: tr.Duration(),
+			Logs:      taskTestLogURL{tr.LogURL},
 		}
-		destTask.LocalTestResults[tr.TestFile] = testResult
+		destTask.LocalTestResults[tr.TestName] = testResult
 	}
```

Comment:
will `srcTask.LocalTestResults` be populated?

Reply:
Yep! On L146 above.